### PR TITLE
details of SM2

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by chaoticlonghair** [comment]
Thanks!

I will fix the Travis CI, then update and merge this PR.

File: docs/sm2.md [hunk 1]
```diff
@@ -74,3 +74,41 @@ let der = signature.der_encode();
...
+
+Finally, `e` is verified by SM2 algorithm. See section 6.2 and 7.2 of the second part of [this documentation](http://www.oscca.gov.cn/sca/xxgk/2010-12/17/1002386/files/b791a9f908bb4803875ab6aeeb7b4e03.pdf).
+
+```
+isValid = SM2_Verify(e, signature, publicKey)
+```
+
+Signatures generated by this library is compatible with [GmSSL](https://github.com/guanzhi/GmSSL).
+
+Be careful, in SM2, we **cannot** recover the public key using the message and the signature, like what Ethereum did. Because before the verification, the public key must be provided to calculate `e`. To solve this, append the public key after the signature, and extract it before the verification.o
```
Comment (chaoticlonghair) [inline docs/sm2.md]: The `o` in the end of this line seems to be a typo.

**Comment by chaoticlonghair** [comment]
Thanks for your commits!

And, could you fix the typo and compress your commits, then `push --force` to update this PR?

Next time, you can use `[skip ci]` if your commits only change docs.

**Comment by ghost** [comment]
Fixed the typo and rebased my commits with [skip ci].

**Review by chaoticlonghair — APPROVED** [review]

**Comment by chaoticlonghair** [comment]
Thank you!

**Pull request merged**